gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py

```diff
@@ -4,537 +4,197 @@
+        cm = configurator.ConfiguratorModule(sc)
+        dmx = demuxer.ConfiguratorDemuxer()
+        rpc_mgr = configurator.ConfiguratorRpcManager(sc, cm, conf, dmx)
         return sc, conf, rpc_mgr
 
-    def _get_GenericConfigRpcManager_object(self, conf, sc):
-        agent = gc.GenericConfigRpcManager(sc, conf)
+    def _get_lbaas_rpc_manager_object(self, conf, sc):
+        agent = lb.LBaaSRpcManager(sc, conf)
         return agent, sc
 
-    @mock.patch(__name__ + '.FakeObjects.nqueue')
-    @mock.patch(__name__ + '.FakeObjects.drivers')
-    def _get_GenericConfigEventHandler_object(
-            self, sc, rpcmgr, drivers, nqueue):
-        agent = gc.GenericConfigEventHandler(sc, drivers, rpcmgr, nqueue)
-        return agent
+    def _test_controller(self, operation):
+        sc, conf, rpc_mgr = self._get_configurator_rpc_manager_object()
+        agent, sc = self._get_lbaas_rpc_manager_object(conf, sc)
 
-    def _get_LBaasRpcManager_object(self, conf, sc):
-        agent = lb.LBaaSRpcManager(sc, conf)
-        return agent, sc
+        method = self.fo.method
 
-    def _test_vip_event_creation(self, operation):
-        sc, conf, rpc_mgr = self._get_ConfiguratorRpcManager_object()
-        agent, sc = self._get_LBaasRpcManager_object(conf, sc)
-
-        arg_dict = {'context': self.lb.context,
-                    'vip': self.lb._fake_vip_obj(),
-                    'serialize': True,
-                    'binding_key': '6350c0fd-07f8-46ff-b797-62acd23760de'
-                    }
-        arg_dict_update = {'context': self.lb.context,
-                    'vip': self.lb._fake_vip_obj(),
-                    'old_vip': self.lb._fake_old_vip_obj(),
-                    'serialize': True,
-                    'binding_key': '6350c0fd-07f8-46ff-b797-62acd23760de'
-                    }
-        method = {'CREATE_VIP': 'create_network_device_config',
-                  'DELETE_VIP': 'delete_network_device_config',
-                  'UPDATE_VIP': 'update_network_device_config'
-                  }
         with mock.patch.object(sc, 'new_event', return_value='foo') as (
             mock_sc_new_event), \
             mock.patch.object(sc, 'post_event') as mock_sc_post_event, \
             mock.patch.object(rpc_mgr,
                               '_get_service_agent_instance',
                               return_value=agent):
-            if method[operation] == 'create_network_device_config' or method[operation]=='delete_network_device_config':
-                request_data = self.lb.fake_request_data_vip()
-                args = arg_dict
-            elif method[operation] == 'update_network_device_config':
-                request_data = self.lb.fake_request_data_vip_update()
-                args = arg_dict_update
-                
-            getattr(rpc_mgr, method[operation])(self.lb.context, request_data)
+            if operation == 'CREATE_VIP' or operation == 'DELETE_VIP':
```

Comment:
We don't need these if else blocks if we pass (operation,request_data,args) to _test_controller() from the respective functions.Please change accordingly